### PR TITLE
added: support output of explicit function to VTF

### PR DESCRIPTION
**Pull request by akva2**
Useful for inspection purposes

### CODE REVIEW

File: src/SIM/SIMoutput.C [hunk 1]
```diff
@@ -83,9 +86,23 @@ bool SIMoutput::parseOutputTag (const TiXmlElement* elem)
 {
   IFEM::cout <<"  Parsing <"<< elem->Value() <<">"<< std::endl;
 
-  if (strcasecmp(elem->Value(),"resultpoints"))
+  if (strcasecmp(elem->Value(),"resultpoints") && strcasecmp(elem->Value(),"function"))
     return this->SIMinput::parseOutputTag(elem);
 
+  if (!strcasecmp(elem->Value(),"function")) {
```
Comment (kmokstad) [inline src/SIM/SIMoutput.C]: A better logic would be:
```
  if (!strcasecmp(elem->Value(),"function"))
  {
    ...do the new function stuff...
  }
  else if (strcasecmp(elem->Value(),"resultpoints"))
    return this->SIMinput::parseOutputTag(elem);

  ...do the rest...
```
and you don't need to check for "function" tag more than once

File: src/SIM/SIMoutput.C [hunk 2]
```diff
@@ -513,8 +530,11 @@ bool SIMoutput::writeGlvS (const Vector& psol, int iStep, int& nBlock,
 {
   idBlock = this->writeGlvS1(psol,iStep,nBlock,time,
                              pvecName,idBlock,psolComps);
+
   if (idBlock < 0)
     return false;
+  else if (!this->writeAddFuncs(iStep,nBlock,50,time))
```
Comment (kmokstad) [inline src/SIM/SIMoutput.C]: Careful with the idBlock value here (50). Ensure it does not conflict with others.

File: src/SIM/SIMoutput.C [hunk 3]
```diff
@@ -1801,3 +1821,18 @@ bool SIMoutput::serialize (std::map<std::string,std::string>&) const
             <<"     Restart not supported for "<< this->getName() << std::endl;
   return false;
 }
+
+
+bool SIMoutput::writeAddFuncs(int iStep, int& nBlock, int idBlock, double time)
+{
+  if (myAddScalars.empty())
+    return true;
```
Comment (kmokstad) [inline src/SIM/SIMoutput.C]: This check is not needed. If empty, the loop below will handle it correctly?

File: src/SIM/SIMoutput.C [hunk 4]
```diff
@@ -1801,3 +1821,18 @@ bool SIMoutput::serialize (std::map<std::string,std::string>&) const
...
+
+  for (const auto& func : myAddScalars) {
+    if (!this->writeGlvF(*func.second, func.first.c_str(), iStep,
+                         nBlock, idBlock, time))
+        return false;
+  }
```
Comment (kmokstad) [inline src/SIM/SIMoutput.C]: Braces are not needed since just single-statement (cosmetic)

File: src/SIM/SIMoutput.h [hunk 5]
```diff
@@ -314,6 +314,9 @@ class SIMoutput : public SIMinput
                          int& nBlock, std::vector< std::vector<int> >& sID,
                          size_t* i = nullptr);
 
+  //! \brief Private helper to write additional functions to VTF-file.
+  bool writeAddFuncs(int iStep, int& nBlock, int idBlock = 50, double time = 0.0);
```
Comment (kmokstad) [inline src/SIM/SIMoutput.h]: Since it is a private helper, it does not need default values either.

File: src/SIM/SIMoutput.C (hunk 1 above)
Comment (akva2) [reply src/SIM/SIMoutput.C]: okay

File: src/SIM/SIMoutput.C (hunk 4 above)
Comment (akva2) [reply src/SIM/SIMoutput.C]: ok

File: src/SIM/SIMoutput.C (hunk 3 above)
Comment (akva2) [reply src/SIM/SIMoutput.C]: indeed.

File: src/SIM/SIMoutput.C (hunk 2 above)
Comment (akva2) [reply src/SIM/SIMoutput.C]: 50 is the default from writeGlvF, but i now make sure to increase it for each registered function (starting at 50).

File: src/SIM/SIMoutput.h (hunk 5 above)
Comment (akva2) [reply src/SIM/SIMoutput.h]: indeed

**Pull request merged**